cpp/src/generators/erdos_renyi_generator.cu

```diff
@@ -42,45 +41,38 @@ generate_erdos_renyi_graph_edgelist_gnp(raft::handle_t const& handle,
                   "Implementation cannot support specified value");
 
-  auto random_iterator = thrust::make_transform_iterator(
-    thrust::make_counting_iterator<size_t>(0),
-    cuda::proclaim_return_type<float>([seed] __device__(size_t index) {
-      thrust::default_random_engine rng(seed);
-      thrust::uniform_real_distribution<float> dist(0.0, 1.0);
-      rng.discard(index);
-      return dist(rng);
-    }));
+  size_t max_num_edges = static_cast<size_t>(num_vertices) * num_vertices;
 
-  size_t count = thrust::count_if(handle.get_thrust_policy(),
-                                  random_iterator,
-                                  random_iterator + num_vertices * num_vertices,
-                                  [p] __device__(float prob) { return prob < p; });
-
-  rmm::device_uvector<size_t> indices_v(count, handle.get_stream());
+  auto generate_random_value = cuda::proclaim_return_type<float>([seed] __device__(size_t index) {
+    thrust::default_random_engine rng(seed);
+    thrust::uniform_real_distribution<float> dist(0.0, 1.0);
+    rng.discard(index);
+    return dist(rng);
+  });
 
-  thrust::copy_if(handle.get_thrust_policy(),
-                  random_iterator,
-                  random_iterator + num_vertices * num_vertices,
-                  indices_v.begin(),
-                  [p] __device__(float prob) { return prob < p; });
+  size_t count = thrust::count_if(handle.get_thrust_policy(),
+                                  thrust::make_counting_iterator<size_t>(0),
+                                  thrust::make_counting_iterator<size_t>(max_num_edges),
+                                  [generate_random_value, p] __device__(size_t index) {
+                                    return generate_random_value(index) < p;
+                                  });
 
   rmm::device_uvector<vertex_t> src_v(count, handle.get_stream());
   rmm::device_uvector<vertex_t> dst_v(count, handle.get_stream());
 
-  thrust::transform(handle.get_thrust_policy(),
-                    indices_v.begin(),
-                    indices_v.end(),
-                    thrust::make_zip_iterator(thrust::make_tuple(src_v.begin(), src_v.end())),
+  thrust::copy_if(handle.get_thrust_policy(),
+                  thrust::make_counting_iterator<size_t>(0),
+                  thrust::make_counting_iterator<size_t>(max_num_edges),
+                  thrust::make_transform_output_iterator(
+                    thrust::make_zip_iterator(src_v.begin(), dst_v.begin()),
                     cuda::proclaim_return_type<thrust::tuple<vertex_t, vertex_t>>(
                       [num_vertices] __device__(size_t index) {
-                        size_t src = index / num_vertices;
-                        size_t dst = index % num_vertices;
-
-                        return thrust::make_tuple(static_cast<vertex_t>(src),
-                                                  static_cast<vertex_t>(dst));
-                      }));
-
-  handle.sync_stream();
```

Comment:
Is it OK to leave out handle.sync_stream() here, as there is one after the kernel call in cpp/tests/generators/erdos_renyi_test.cpp?

Reply:
Yes.  The `sync_stream` call here was never necessary.  Any asynchronous work that is active on this stream will continue to be active when we leave this function call.